### PR TITLE
Rename initialBlock to defaultBlock

### LINKED ISSUE #993: EditorConfig: rename `initialBlock` to the `defaultBlock`
Because current naming causes misunderstanding with the real purpose of this option.

See #922

### PR DESCRIPTION
**Pull request by ranemihir**
Closes #993 

The initialBlock property is renamed to defaultBlock.

### CODE REVIEW

File: src/components/core.ts [hunk 1]
```diff
@@ -158,9 +158,9 @@ export default class Core {
     _.setLogLevel(this.config.logLevel);
 
     /**
-     * If initial Block's Tool was not passed, use the Paragraph Tool
+     * If default Block's Tool was not passed, use the Paragraph Tool
      */
-    this.config.initialBlock = this.config.initialBlock || 'paragraph';
+    this.config.defaultBlock = this.config.defaultBlock || 'paragraph';
```
Comment (neSpecc) [inline src/components/core.ts:163]: we need to save the previous deprecated name for backward capability issues

Comment (ranemihir) [reply src/components/core.ts:163]: I do as u said, try to keep both the `initialBlock` and the `defaultBlock` and push a new commit ASAP.

File: src/components/utils.ts [hunk 2]
```diff
@@ -556,3 +556,18 @@ export function getValidUrl(url: string): string {
 export function openTab(url: string): void {
   window.open(url, '_blank');
 }
+
+/**
+ * Common method for deprecating properties or methods.
```
Comment (neSpecc) [inline src/components/utils.ts:561]:
```suggestion
 * Common method for printing a warning about the usage of deprecated property or method.
```

File: src/components/utils.ts [hunk 3]
```diff
@@ -556,3 +556,18 @@ export function getValidUrl(url: string): string {
...
+ * 
+ * @param condition - condition for deprecation.
+ * @param oldProperty - deprecated property.
+ * @param newProperty - use this property instead of deprecated property.
```
Comment (neSpecc) [inline src/components/utils.ts:565]:
```suggestion
 * @param newProperty - the property that should be used instead.
```

File: src/components/utils.ts [hunk 4]
```diff
@@ -556,3 +556,18 @@ export function getValidUrl(url: string): string {
...
+ */
+export function deprecationAssert(condition: boolean, oldProperty: string, newProperty: string) {
+  const message = `${oldProperty} is deprecated and will be removed in the next major release. Please use ${newProperty} instead.`;
```
Comment (neSpecc) [inline src/components/utils.ts:568]:
```suggestion
  const message = `«${oldProperty}» is deprecated and will be removed in the next major release. Please use the «${newProperty}» instead.`;
```

File: types/configs/editor-config.d.ts [hunk 5]
```diff
@@ -25,6 +25,12 @@ export interface EditorConfig {
    * Name should be equal to one of Tool`s keys of passed tools
    * If not specified, Paragraph Tool will be used
    */
+  defaultBlock?: string;
+
+  /**
+   * This property will be depracated in the next major relase.
```
Comment (neSpecc) [inline types/configs/editor-config.d.ts:31]:
```suggestion
   * @deprecated
   * This property will be deprecated in the next major release. 
```

File: types/configs/editor-config.d.ts [hunk 6]
```diff
@@ -25,6 +25,12 @@ export interface EditorConfig {
    * Name should be equal to one of Tool`s keys of passed tools
    * If not specified, Paragraph Tool will be used
    */
+  defaultBlock?: string;
+
+  /**
+   * This property will be depracated in the next major relase. 
+   * Use 'defaultBlock' property instead.
```
Comment (neSpecc) [inline types/configs/editor-config.d.ts:32]:
```suggestion
   * Use the 'defaultBlock' property instead.
```

File: src/components/modules/blockManager.ts [hunk 7]
```diff
@@ -649,15 +649,15 @@ export default class BlockManager extends Module {
   /**
    * Clears Editor
    *
-   * @param {boolean} needAddInitialBlock - 1) in internal calls (for example, in api.blocks.render)
+   * @param {boolean} needAddDefaultBlock - 1) in internal calls (for example, in api.blocks.render)
    *                                        we don't need to add empty initial block
    *                                        2) in api.blocks.clear we should add empty block
    */
-  public clear(needAddInitialBlock = false): void {
+  public clear(needAddDefaultBlock = false): void {
```
[the remark this answers is not in the document]
Comment (ranemihir) [reply src/components/modules/blockManager.ts]: Done.

**Pull request merged**